bindings/matrix-sdk-crypto-ffi/src/machine.rs

```diff
@@ -560,6 +560,100 @@ impl OlmMachine {
             .map(|r| r.into()))
     }
 
+    /// Get the stored room settings, such as the encryption algorithm or
+    /// whether to encrypt only for trusted devices.
+    ///
+    /// These settings can be modified via
+    /// [set_room_algorithm()](#method.set_room_algorithm) and
+    /// [set_room_only_allow_trusted_devices()](#method.
+    /// set_room_only_allow_trusted_devices) methods.
+    pub fn get_room_settings(
+        &self,
+        room_id: &str,
+    ) -> Result<Option<RoomSettings>, CryptoStoreError> {
+        let room_id = RoomId::parse(room_id)?;
+        let settings = self
+            .runtime
+            .block_on(self.inner.store().get_room_settings(&room_id))?
+            .map(|v| v.try_into().unwrap());
```

Comment:
How can I flatten the possible error from `try_into` and return as `CryptoStoreError`?